### PR TITLE
fixes for Windows cpp17

### PR DESCRIPTION
**Pull request by bam241**
filesystem::canonical() needs to be converted to char* ...

### CODE REVIEW

**Comment by bam241** [comment]
`const char* full_filepath = std::filesystem::canonical(filename.c_str()).string().c_str();` could have been directly casted as a string:
`string full_filepath = std::filesystem::canonical(filename.c_str()).string();`
but I kept it that way for consistency, but can change it if anyone feels strongly about this...

File: src/utils.cpp [hunk 1]
```diff
@@ -393,7 +393,7 @@ std::string pyne::get_full_filepath(std::string filename) {
 #ifndef _WIN32
   const char* full_filepath = realpath(filename.c_str(), NULL);
 #else
-  const char* full_filepath = std::filesystem::canonical(filename.c_str());
+  const char* full_filepath = std::filesystem::canonical(filename.c_str()).string().c_str();
```
Comment (gonuke) [inline src/utils.cpp]: We are about to recast this as a `std::string`.  Can't we just make `full_filepath` be `std::string` and avoid the conversion to `char*`?

**Review by gonuke — CHANGES_REQUESTED** [review]
I thought I had already submitted this review (my iPad app is throwing me off)...

and now I see that you already recognized the same thing.  I think it makes sense to define `full_filepath` as a `std::string` in both instances.  The results of `realpath` will be promoted to the `std::string` on assignment.

File: src/utils.cpp (hunk 1 above)
Comment (bam241) [reply src/utils.cpp]: yes that's what I was suggesting. I kept it that way so no windows and windows have the same kind of behavior and the temp variable has the same type...

**Review by gonuke — APPROVED** [review]
LGTM - thanks @bam241

**Pull request merged**